db/python/tables/analysis.py

```diff
@@ -255,8 +248,32 @@ async def query(self, filter_: AnalysisFilter) -> List[AnalysisInternal]:
         """
         rows = await self.connection.fetch_all(_query, values)
         result: list[AnalysisInternal] = []
+
+        if not rows:
+            return result
+
+        analysis_outputs_by_aid = await self.get_file_outputs_by_analysis_ids(
+            [r['id'] for r in rows]
+        )
+
         for row in rows:
-            analysis = AnalysisInternal.from_db(**dict(row))
+
+            analysis_data = dict(row)
+            analysis_output_for_id = analysis_outputs_by_aid.get(
+                analysis_data['id'], None
+            )
+            analysis_data['output'] = (
+                analysis_output_for_id.get('output', None)
+                if analysis_output_for_id
```

Comment:
The AnalysisInternal model from_db method sets the defaults for these, so you could get away with a single 

```python
if analysis_output_for_id:
```

statement eg. 

```python

if analysis_output_for_id:
    analysis_data['output'] = analysis_output_for_id.get('output', None)
    analysis_data['outputs'] = analysis_output_for_id.get('outputs', {})

```

Reply:
Good point!